src/flow_solver/flow_solver_cases/periodic_1D_unsteady.cpp

```diff
@@ -18,18 +18,19 @@ Periodic1DUnsteady<dim, nstate>::Periodic1DUnsteady(const PHiLiP::Parameters::Al
 
 
 template <int dim, int nstate>
-double Periodic1DUnsteady<dim, nstate>::compute_energy_collocated(
+double Periodic1DUnsteady<dim, nstate>::compute_energy(
         const std::shared_ptr <DGBase<dim, double>> dg
         ) const
 {
-    // Intention is to eventually calculate energy in physics and generalize to arbitrary nodes
-    // For now, using collocated energy calculation
-    double energy = 0.0;
-    for (unsigned int i = 0; i < dg->solution.size(); ++i)
-    {
-        energy += 1./(dg->global_inverse_mass_matrix.diag_element(i)) * dg->solution(i) * dg->solution(i);
-    }
-    return energy;
+    //Calculating energy via matrix-vector product
+    dealii::LinearAlgebra::distributed::Vector<double> temp;
+    temp.reinit(dg->solution);
+    if(this->all_param.use_inverse_mass_on_the_fly){
+        dg->apply_global_mass_matrix(dg->solution, temp);
```

Comment:
If we are running with FR, would this allow the option of computing the energy with a different norm?

Reply:
This function computes energy as `u^T M_FR u`. 

The `apply_global_mass_matrix` function applies the modified mass matrix (i.e., \tilde{M} = M + K). Therefore, energy is computed in the correct broken Sobolev norm. For Burgers, I observe conservation even in c != cDG cases.

I would have to look at other parts of the code to see how to compute energy in different norms.